NPM update issues

@tomjn now xedug traces are not created automatically anymore, but I got this series of errors when provisioning

```
default: Cleaning apt caches...
    default: Installing/updating npm...
    default: npm
    default:  ERR! path /usr/bin/npm
    default: npm ERR! code EACCES
    default: npm
    default:
    default: ERR! errno -13
    default: npm ERR! syscall unlink
    default: npm ERR! Error: EACCES: permission denied, unlink '/usr/bin/npm'
    default: npm ERR!  { [OperationalError: EACCES: permission denied, unlink '/usr/bin/npm']
    default: npm ERR!   cause:
    default: npm ERR!    { [Error: EACCES: permission denied, unlink '/usr/bin/npm']
    default: npm ERR!
    default:       errno: -13,
    default: npm ERR!      code: 'EACCES',
    default: npm ERR!      syscall: 'unlink',
    default: npm ERR!      path: '/usr/bin/npm' },
    default: npm ERR!   stack: "Error: EACCES: permission denied, unlink '/usr/bin/npm'",
    default: npm ERR!   errno: -13,
    default: npm ERR!
    default:    code: 'EACCES',
    default: npm
    default: ERR!   syscall: 'unlink',
    default: npm
    default:  ERR!   path: '/usr/bin/npm' }
    default: npm
    default: ERR!
    default: npm ERR! The operation was rejected by your operating system.
    default: npm ERR! It is likely you do not have the permissions to access this file as the current user
    default: npm ERR!
    default: npm ERR! If you believe this might be a permissions issue, please double-check the
    default: npm
    default:  ERR! permissions of the file and its containing directories, or try running
    default: npm ERR! the command again as root/Administrator (though this is not recommended).
    default: npm ERR! A complete log of this run can be found in:
    default: npm ERR!     /home/vagrant/.npm/_logs/2019-06-20T14_28_40_849Z-debug.log
    default: Installing/updating npm-check-updates...
    default: npm
    default:  ERR! path /usr/bin/ncu
    default: npm
    default:
    default: ERR! code EACCES
    default: npm
    default:
    default: ERR! errno
    default:  -13
    default: npm
    default:  ERR! syscall unlink
    default: npm
    default:  ERR! Error: EACCES: permission denied, unlink '/usr/bin/ncu'
    default: npm ERR!  { [OperationalError: EACCES: permission denied, unlink '/usr/bin/ncu']
    default: npm ERR!   cause:
    default: npm ERR!    { [Error: EACCES: permission denied, unlink '/usr/bin/ncu']
    default: npm
    default: ERR!      errno: -13,
    default: npm ERR!
    default:       code: 'EACCES',
    default: npm ERR!
    default:       syscall: 'unlink',
    default: npm
    default: ERR!      path: '/usr/bin/ncu' },
    default: npm
    default: ERR!
    default:    stack: "Error: EACCES: permission denied, unlink '/usr/bin/ncu'",
    default: npm ERR!   errno: -13,
    default: npm
    default: ERR!   code: 'EACCES',
    default: npm
    default: ERR!
    default:    syscall: 'unlink',
    default: npm ERR!   path: '/usr/bin/ncu' }
    default: npm
    default:  ERR!
    default:
    default: npm ERR!
    default:  The operation was rejected by your operating system.
    default: npm ERR!
    default:  It is likely you do not have the permissions to access this file as the current user
    default: npm ERR!
    default: npm ERR! If you believe this might be a permissions issue, please double-check the
    default: npm
    default: ERR!
    default:  permissions of the file and its containing directories, or try running
    default: npm ERR! the command again as root/Administrator (though this is not recommended).
    default: npm ERR!
    default:  A complete log of this run can be found in:
    default: npm ERR!     /home/vagrant/.npm/_logs/2019-06-20T14_28_52_186Z-debug.log
    default: ack-grep already installed
```

_Originally posted by @kmgalanakis in https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1866#issuecomment-504051848_

--- a/provision/provision.sh
+++ b/provision/provision.sh
@@ -437,6 +437,9 @@ tools_install() {
   # Disable xdebug before any composer provisioning.
   sh /srv/config/homebin/xdebug_off
 
+  if [[ -f ~/.nvm ]]; then
+    rm -rf ~/.nvm ~/.npm ~/.bower
+  fi
   # npm
   #
   # Make sure we have the latest npm version and the update checker module

--- a/provision/provision.sh
+++ b/provision/provision.sh
@@ -438,7 +438,7 @@ tools_install() {
   sh /srv/config/homebin/xdebug_off
 
   if [[ -f ~/.nvm ]]; then
-    rm -rf ~/.nvm ~/.npm ~/.bower
+    rm -rf ~/.nvm ~/.npm ~/.bower /srv/config/nvm
   fi
   # npm
   #

--- a/provision/provision.sh
+++ b/provision/provision.sh
@@ -397,7 +397,7 @@ package_install() {
     apt-key add /srv/config/apt-keys/mongo-server-4.0.asc
   fi
 
-  # Update all of the package references before installing anything
+  # Update all of the package references before installing anythin
   echo "Running apt-get update..."
   apt-get -y update
 
@@ -437,8 +437,12 @@ tools_install() {
   # Disable xdebug before any composer provisioning.
   sh /srv/config/homebin/xdebug_off
 
+  echo "Checking for NVM"
   if [[ -f ~/.nvm ]]; then
+    echo ".nvm folder found, switching to system node, and removing NVM folders"
+    nvm use system
     rm -rf ~/.nvm ~/.npm ~/.bower /srv/config/nvm
+    echo "NVM folders removed"
   fi
   # npm
   #
@@ -459,15 +463,15 @@ tools_install() {
     curl -s https://beyondgrep.com/ack-2.16-single-file > "/usr/bin/ack" && chmod +x "/usr/bin/ack"
   fi
 
-  # Make sure the composer cache is not owned by root
+  echo "Making sure the composer cache is not owned by root"
   mkdir -p /usr/local/src/composer
   mkdir -p /usr/local/src/composer/cache
   chown -R vagrant:www-data /usr/local/src/composer
   chown -R vagrant:www-data /usr/local/bin
 
   # COMPOSER
-  #
-  # Install Composer if it is not yet available.
+
+  echo "Checking for Composer"
   exists_composer="$(which composer)"
   if [[ "/usr/local/bin/composer" != "${exists_composer}" ]]; then
     echo "Installing Composer..."
@@ -477,6 +481,7 @@ tools_install() {
   fi
 
   if [[ -f /srv/provision/github.token ]]; then
+    echo "A personal GitHub token was found, configuring composer"
     ghtoken=`cat /srv/provision/github.token`
     noroot composer config --global github-oauth.github.com $ghtoken
     echo "Your personal GitHub token is set for Composer."
@@ -486,9 +491,9 @@ tools_install() {
   # the master branch on its GitHub repository.
   if [[ -n "$(noroot composer --version --no-ansi | grep 'Composer version')" ]]; then
     echo "Updating Composer..."
+    COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global config bin-dir /usr/local/bin
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi self-update --no-progress --no-interaction
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global require --no-update --no-progress --no-interaction phpunit/phpunit:6.* phpunit/php-invoker:1.1.* mockery/mockery:0.9.* d11wtq/boris:v1.0.8
-    COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global config bin-dir /usr/local/bin
     COMPOSER_HOME=/usr/local/src/composer noroot composer --no-ansi global update --no-progress --no-interaction
   fi
 

--- a/provision/provision.sh
+++ b/provision/provision.sh
@@ -397,7 +397,7 @@ package_install() {
     apt-key add /srv/config/apt-keys/mongo-server-4.0.asc
   fi
 
-  # Update all of the package references before installing anythin
+  # Update all of the package references before installing anything
   echo "Running apt-get update..."
   apt-get -y update
 
@@ -448,9 +448,9 @@ tools_install() {
   #
   # Make sure we have the latest npm version and the update checker module
   echo "Installing/updating npm..."
-  noroot npm install -g npm
+  npm install -g npm
   echo "Installing/updating npm-check-updates..."
-  noroot npm install -g npm-check-updates
+  npm install -g npm-check-updates
 
   # ack-grep
   #

